node_test.go

```diff
@@ -333,3 +368,51 @@ func TestExamples(t *testing.T) {
 		}
 	}
 }
+
+func TestCanonicalize(t *testing.T) {
+	b, err := ioutil.ReadFile("test_objects/non-canon.cbor")
+	if err != nil {
+		t.Fatal(err)
+	}
+	nd1, err := Decode(b)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if bytes.Equal(b, nd1.RawData()) {
+		t.Fatal("failed to canonicalize node")
+	}
+	nd2, err := Decode(nd1.RawData())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !nd2.Cid().Equals(nd1.Cid()) || !bytes.Equal(nd2.RawData(), nd1.RawData()) {
+		t.Fatal("re-decoding a canonical node should be idempotent")
+	}
+}
```

Comment:
If test_objects/non-canon.cbor is taken from go-ipfs I'd add a check for whether CID is `zdpuAmxF8q6iTUtkB3xtEYzmc5Sw762qwQJftt5iW8NTWLtjC` here